src/Andre/SoulsFormats/SoulsFormats/Util/BinaryReaderEx.cs

```diff
@@ -1136,6 +1136,44 @@ public string GetShiftJIS(long offset, int length)
             return result;
         }
 
+        /// <summary>
+        /// Reads a null-terminated UTF-8 string.
+        /// </summary>
+        public string ReadUTF8()
+        {
+            List<byte> bytes = new List<byte>(64);
+            byte a = ReadByte();
+            byte b = ReadByte();
+            while (a != 0 || b != 0)
+            {
+                bytes.Add(a);
+                bytes.Add(b);
+                a = ReadByte();
+                b = ReadByte();
+            }
+
+            return SFEncoding.UTF8.GetString(bytes.ToArray());
+        }
+
+        /// <summary>
+        /// Reads a CR+LF terminated UTF-8 string.
+        /// </summary>
+        public string ReadUTF8Line()
+        {
+            List<byte> bytes = new List<byte>(64);
+            byte a = ReadByte();
+            byte b = ReadByte();
+            while (a != 13 && b != 10)
+            {
+                bytes.Add(a);
+                bytes.Add(b);
+                a = ReadByte();
+                b = ReadByte();
```

Comment:
Similar to above, I'd recommend reading one byte at a time, checking for end-of-input, and simply shuffling b into a each iteration. I'd also try find a named enumeration of crlf instead of leaving magic numbers in there (they'll compile out if they're constants anyway, but it keeps it cleaner.)

Reply:
Same answer as above, replaced magic number with \r.